### ISSUE TITLE
Database Huge - Needs removal of rows.

### ISSUE THREAD

**Issue by Flumadiddle** [AHAAAAAAA/PokemonGo-Map#3264]
I have 300 accounts scanning the city and after a few days, the MySQL database has reached over 2 million Pokémon records and querying has become insanely slow.

Anyway to implement autoremoval of the row once the Pokémon expires?

**Comment by Sharke** [comment]
I'm personally glad it stores it. It allows my to query against it for spawn patterns. Why not simply just wipe the database?

**Comment by andrewda** [comment]
How about we add a CLI argument for whether or not the database rows are removed after the Pokemon despawns?

**Comment by SOLiNARY** [comment]
If u r lazy, just keep cron job which will run mysql & truncate all tables each day.

**Comment by Flumadiddle** [comment]
Yeah, a cronjob was a temporary workaround although truncating would mean the map will become empty for a few minutes. 

CLI argument would be the ideal solution.

**Comment by ColbyMiner** [comment]
Another idea is to not store common Pokemon in the DB in the first place. In my case, I filter these all out anyway in the map.

**Comment by shiona** [comment]
@Flumadiddle If possible could you please save that big database and possibly share it? 
I believe it could be useful in profiling.

I mean with the index on disappear_time the fetch for current pokemon should be a binary search, which for a few million rows shouldn't be any kind of problem.

**Comment by vemfanvet** [comment]
Where is the db stored? My folder is starting to take up a lot of space and I want to know what folder/file to delete

**Comment by algoni** [comment]
@vemfanvet pogom.db if using sqlite (default).

**Comment by vemfanvet** [comment]
@algoni that file only takes 2mb while the node_module folder takes 200mb.

**Comment by algoni** [comment]
Well that's still the database file ;) node_modules is a folder for, well, node.js modules that are needed on the dev branch and are created when you run `npm install`.

**Comment by vemfanvet** [comment]
@algoni alright, will it up less space if I delete node_module and run npm install again?

**Comment by algoni** [comment]
Nope, running npm install will just reinstall those packages.

What you *can* do, however, is `npm install` and `grunt build` to create the static assets, then you can safely delete the node_modules folder, until you need to run `grunt build` again.

**Comment by Sharke** [comment]
I mean... You could truncate yourself if they're despawned.

Just compare the time.
